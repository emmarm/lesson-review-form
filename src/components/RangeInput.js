import React from 'react';
import PropTypes from 'prop-types';

const RangeInput = ({ input, className, inputId, label, title }) => (
  <div className={className}>
    <label className={className + '__label'} htmlFor={inputId} title={title}>
      {label}
    </label>
    <input
      {...input}
      id={inputId}
      className={className + '__input'}
      type="range"
      min={0}
      max={10}
      step={1}
      value={input.value || '0'}
    />
    <div className={className + '__display'}>{input.value || '0'}</div>
  </div>
);

RangeInput.propTypes = {
  className: PropTypes.string.isRequired,
  inputId: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired
};

export default RangeInput;
