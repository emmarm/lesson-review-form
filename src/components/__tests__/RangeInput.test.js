import React from 'react';
import { shallow } from 'enzyme';

import RangeInput from '../RangeInput';

describe('RangeInput', () => {
  const props = {
    input: { value: '' },
    className: 'dummy',
    inputId: 'dummy',
    label: 'Dummy Input'
  };
  const wrapper = shallow(<RangeInput {...props} />);

  it('renders RangeInput snapshot correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('correctly applies class names passed through props', () => {
    const container = wrapper.find('div').first();
    const label = wrapper.find('label').first();
    const input = wrapper.find('input').first();
    const display = wrapper.find('div').last();

    expect(container.hasClass('dummy')).toBeTruthy();
    expect(label.hasClass('dummy__label')).toBeTruthy();
    expect(input.hasClass('dummy__input')).toBeTruthy();
    expect(display.hasClass('dummy__display')).toBeTruthy();
  });

  it('applies value 0 when undefined value is passed in', () => {
    const input = wrapper.find('input').first();
    const display = wrapper.find('div').last();
    expect(input.props().value).toBe('0');
    expect(display.text()).toBe('0');
  });
});
