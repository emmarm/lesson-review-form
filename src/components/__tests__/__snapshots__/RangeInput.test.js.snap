// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeInput renders RangeInput snapshot correctly 1`] = `
<div
  className="dummy"
>
  <label
    className="dummy__label"
    htmlFor="dummy"
  >
    Dummy Input
  </label>
  <input
    className="dummy__input"
    id="dummy"
    max={10}
    min={0}
    step={1}
    type="range"
    value="0"
  />
  <div
    className="dummy__display"
  >
    0
  </div>
</div>
`;
